app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,481 @@
+            const buttons = {
+                rightButtons: [{
+                    ...rightButton!,
+                    enabled: value,
+                }],
+            };
+            setButtons(componentId, buttons);
+        }
+        setCanSave(value);
+    }, [componentId, rightButton]);
+
+    const submitUser = useCallback(preventDoubleTap(async () => {
+        enableSaveButton(false);
+        setError(undefined);
+        setUpdating(true);
+        try {
+            const partialUser: Partial<UserProfile> = {
+                email: userInfo.email,
+                first_name: userInfo.firstName,
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                resetScreen(reqError as Error);
+                return;
+            }
+
+            close();
+        } catch (e) {
+            resetScreen(e as Error);
+        }
+    }), [userInfo]);
+
+    const resetScreen = useCallback((resetError: Error) => {
+        setError(resetError?.message);
+        Keyboard.dismiss();
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    }, [enableSaveButton]);
+
+    const updateField = useCallback((fieldKey: string, name: string) => {
+        const update = {...userInfo};
+        update[fieldKey] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[fieldKey];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo, currentUser]);
+
+    const includesSsoService = (sso: string) => ['gitlab', 'google', 'office365'].includes(sso);
+    const isSAMLOrLDAP = (protocol: string) => ['ldap', 'saml'].includes(protocol);
+
+    const userProfileFields: FieldSequence = {
+        firstName: {
+            ref: firstNameRef,
+            isDisabled: (isSAMLOrLDAP(service) && lockedFirstName) || includesSsoService(service),
+        },
+        lastName: {
+            ref: lastNameRef,
+            isDisabled: (isSAMLOrLDAP(service) && lockedLastName) || includesSsoService(service),
+        },
+        username: {
+            ref: usernameRef,
+            isDisabled: service !== '',
+        },
+        email: {
+            ref: emailRef,
+            isDisabled: true,
+        },
+        nickname: {
+            ref: nicknameRef,
+            isDisabled: isSAMLOrLDAP(service) && lockedNickname,
+        },
+        position: {
+            ref: positionRef,
+            isDisabled: isSAMLOrLDAP(service) && lockedPosition,
+        },
+    };
+
+    const hasDisabledFields = Object.values(userProfileFields).filter((field) => field.isDisabled);
```

Comment:
The name implies boolean and the only use is "as a boolean", so... why not make the `> 0` here and have this as a boolean instead of a list?

Reply:
Refactored